packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx

```diff
@@ -204,6 +221,80 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
     }
   }
 
+  private _findCornerIndexes(week: IDayInfo[], weekType: string) {
+    let { firstDayOfWeek, navigatedDate, dateRangeType } = this.props;
+    let cornerIndexes = [];
+
+    if (weekType === 'firstWeek') {
+
+      if (week[DAYS_IN_WEEK - 2].isInMonth) {
+        let firstDayOfMonth = addDays(navigatedDate, - navigatedDate.getDate() + 1);
+        // Account for firstDayOfWeek not beeing Sunday
+        let firstDayIndex = firstDayOfMonth.getDay() - firstDayOfWeek;
+        firstDayIndex = firstDayIndex < 0 ? firstDayIndex + DAYS_IN_WEEK : firstDayIndex;
+        cornerIndexes.push(firstDayIndex);
+      }
+      cornerIndexes.push(DAYS_IN_WEEK - 1);
+    } else {
+      if (week[1].isInMonth) {
+        let daysInMonth = new Date(navigatedDate.getFullYear(), navigatedDate.getMonth() + 1, 0).getDate();
+        let lastDayofMonth = addDays(navigatedDate, daysInMonth - navigatedDate.getDate());
+        // Account for firstDayOfWeek not beeing Sunday
+        let lastDayIndex = lastDayofMonth.getDay() - firstDayOfWeek;
+        lastDayIndex = lastDayIndex < 0 ? lastDayIndex + DAYS_IN_WEEK : lastDayIndex;
+        cornerIndexes.push(lastDayIndex);
+      }
+
+      cornerIndexes.push(0);
+    }
+
+    return cornerIndexes;
+  }
```

Comment:
This function can be vastly simplified to something like this:
  private _findCornerIndexes(week: IDayInfo[]) {
    let cornerIndexes = [];
    for (let i = 0, length = week.length; i < length; i++) {
      let day = week[i];
      if (day.isInMonth) {
        cornerIndexes.push(i);
      }
    }
    if (cornerIndexes.length > 2) {
      cornerIndexes.splice(1, cornerIndexes.length - 2);
    }
    return cornerIndexes;
  }